src/mcmc/gibbs.jl

```diff
@@ -1,3 +1,23 @@
+"""
+    isgibbscomponent(alg::Union{InferenceAlgorithm, AbstractMCMC.AbstractSampler})
+
+Return a boolean indicating whether `alg` is a valid component for a Gibbs sampler.
+
+Defaults to `false` if no method has been defined for a particular algorithm type.
+"""
+isgibbscomponent(::InferenceAlgorithm) = false
+isgibbscomponent(spl::ExternalSampler) = isgibbscomponent(spl.sampler)
+isgibbscomponent(spl::Sampler) = isgibbscomponent(spl.alg)
+
+isgibbscomponent(::ESS) = true
+isgibbscomponent(::HMC) = true
+isgibbscomponent(::HMCDA) = true
+isgibbscomponent(::NUTS) = true
+isgibbscomponent(::MH) = true
+isgibbscomponent(::PG) = true
+isgibbscomponent(::AdvancedHMC.HMC) = true
+isgibbscomponent(::AdvancedMH.MetropolisHastings) = true
```

Comment:
Maybe put these and the `ExternalSampler` definition next to each other since these are externalsamplers?